vaccine/worker.py

```diff
@@ -185,31 +186,36 @@ async def _push_results(self):
         if not answers:
             return
         try:
-            await self.session.post(
+            data = {
+                "data": {
+                    "type": "responses",
+                    "id": config.ANSWER_RESOURCE_ID,
+                    "attributes": {
+                        "responses": [
+                            [
+                                a.timestamp.isoformat(),
+                                a.row_id,
+                                a.address,
+                                a.session_id,
+                                a.question,
+                                a.response,
+                                a.response_metadata,
+                            ]
+                            for a in answers
+                        ]
+                    },
+                }
+            }
+            response = await self.session.post(
                 url=urljoin(
                     config.ANSWER_API_URL,
                     f"flow-results/packages/{config.ANSWER_RESOURCE_ID}/responses/",
                 ),
-                json={
-                    "data": {
-                        "type": "responses",
-                        "id": config.ANSWER_RESOURCE_ID,
-                        "attributes": {
-                            "responses": [
-                                [
-                                    a.timestamp.isoformat(),
-                                    a.row_id,
-                                    a.address,
-                                    a.session_id,
-                                    a.question,
-                                    a.response,
-                                    a.response_metadata,
-                                ]
-                                for a in answers
-                            ]
-                        },
-                    }
-                },
+                json=data,
+            )
+            response_data = await response.json()
+            sentry_sdk.set_context(
+                "answer_api", {"request_data": data, "response_data": response_data}
```

Comment:
I think we'll need to set `raise_for_status=False` where `self.session` is defined, otherwise it will raise the exception before we get to this line. And then manually call to raise for status after we set this context.